src/jira/client/jira-client.ts

```diff
@@ -545,6 +556,6 @@ const updateIssueKeyAssociationValuesFor = (resources: JiraRemoteLink[], mutatin
 };
 
 /**
- * Truncates to 100 elements in an array
+ * Truncates to 500 elements in an array
  */
-const truncate = (array) => array.slice(0, ISSUE_KEY_API_LIMIT);
+const truncate = (array) => array.slice(0, getIssueKeyLimit());
```

Comment:
**IMPORTANT**
This probably don't work, as the `getIssueKeyLimit` return a `Promise`, `array.slice` is expecting a number though.

Please also have a look at the test why this bug is not catched?

Reply:
@gxueatlassian I think it's already being caught because the tests are failing, but I'm rejecting this PR now, it looks like DD is handling the load well.